kopf/k8s/events.py

```diff
@@ -0,0 +1,52 @@
+import datetime
+
+import kubernetes
+
+
+def post_event(*, obj, type, reason, message=''):
+    """
+    Issue an event for the object.
+    """
+
+    now = datetime.datetime.utcnow()
+    namespace = obj['metadata']['namespace']
+
+    # Object reference - similar to the owner reference, but different.
+    # TODO: reconstruct from `resource` once kind<->plural mapping is done. #57
+    ref = dict(
+        apiVersion=obj['apiVersion'],       # resource.version
+        kind=obj['kind'],                   # resource.kind (~resource.plural)
+        name=obj['metadata']['name'],
+        uid=obj['metadata']['uid'],
+        namespace=obj['metadata']['namespace'],
+    )
+
+    meta = kubernetes.client.V1ObjectMeta(
+        namespace=namespace,
+        generate_name='kopf-event-',
+    )
+    body = kubernetes.client.V1beta1Event(
+        metadata=meta,
+
+        action='Action?',
+        type=type,
+        reason=reason,
+        note=message,
+        # message=message,
```

Comment:
are you leaving these comments intentionally? :)

Reply:
Yes. 

I struggle to identify which fields do what here, and they are duplicating each other (but not all of them): e.g. `note` ~ `message`, `regarding` ~ `related`, `event_time` ~ `deprecated_first_timestamp`; also some are of unclear purpose, and are not shown anywhere, e.g. `action`.

I will get back to them later, probably when I will be switching Kopf to pykube-ng — just to see how it is better done with other clients.

This might also be needed for multi-version K8s compatibility, as these fields look like some transitional stage between Kubernetes versions, so it might happen that these lines should be uncommented later. — E.g., most likely, it will be a switch from `V1beta1Event` to something like `V1Event` class.

Reply:
@samurang87 And immediately related: #81 — with the cleanup of these fields (on another occasion).